src/server/scripts/Spells/spell_holiday.cpp

```diff
@@ -1768,6 +1770,144 @@ class spell_midsummer_torch_catch : public SpellScript
+    {
+        return ValidateSpellInfo({ SPELL_FLING_TORCH_TRIGGERED, SPELL_FLING_TORCH_SHADOW });
+    }
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        Position dest = GetCaster()->GetFirstCollisionPosition(30.0f, (float)rand_norm() * static_cast<float>(2 * M_PI));
+        GetCaster()->CastSpell(dest, SPELL_FLING_TORCH_TRIGGERED, true);
+        GetCaster()->CastSpell(dest, SPELL_FLING_TORCH_SHADOW);
+    }
+
+    void Register() override
+    {
+        OnEffectHit += SpellEffectFn(spell_midsummer_fling_torch::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 45669 - Fling Torch
+class spell_midsummer_fling_torch_triggered : public SpellScript
+{
+    PrepareSpellScript(spell_midsummer_fling_torch_triggered);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_JUGGLE_TORCH_MISSED });
+    }
+
+    void HandleTriggerMissile(SpellEffIndex effIndex)
+    {
+        if (Position const* pos = GetHitDest())
+        {
+            if (GetCaster()->GetExactDist2d(pos) > 3.0f)
+            {
+                PreventHitEffect(effIndex);
+                GetCaster()->CastSpell(*GetExplTargetDest(), SPELL_JUGGLE_TORCH_MISSED);
+                GetCaster()->RemoveAura(SPELL_TORCHES_CAUGHT);
+            }
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectHit += SpellEffectFn(spell_midsummer_fling_torch_triggered::HandleTriggerMissile, EFFECT_0, SPELL_EFFECT_TRIGGER_MISSILE);
+    }
+};
+
+// 45671 - Juggle Torch (Catch, Quest)
+class spell_midsummer_fling_torch_catch : public SpellScript
+{
+    PrepareSpellScript(spell_midsummer_fling_torch_catch);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({
+            SPELL_FLING_TORCH_TRIGGERED, SPELL_TORCH_CATCHING_SUCCESS_ALLIANCE, SPELL_TORCH_CATCHING_SUCCESS_HORDE,
+            SPELL_TORCH_CATCHING_REMOVE_TORCHES, SPELL_FLING_TORCH_SHADOW
+        });
+    }
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        Player* player = GetHitPlayer();
+        if (!player)
+            return;
+
+        if (!GetExplTargetDest())
+            return;
+
+        // Only the caster can catch the torch
+        if (player->GetGUID() != GetCaster()->GetGUID())
+            return;
+
+        uint8 requiredCatches = 0;
+        // Number of required catches depends on quest - 4 for the normal quest, 10 for the daily version
+        if (player->GetQuestStatus(QUEST_TORCH_CATCHING_A) == QUEST_STATUS_INCOMPLETE || player->GetQuestStatus(QUEST_TORCH_CATCHING_H) == QUEST_STATUS_INCOMPLETE)
+            requiredCatches = 3;
+        else if (player->GetQuestStatus(QUEST_MORE_TORCH_CATCHING_A) == QUEST_STATUS_INCOMPLETE || player->GetQuestStatus(QUEST_MORE_TORCH_CATCHING_H) == QUEST_STATUS_INCOMPLETE)
+            requiredCatches = 9;
+
+        if (player->GetAuraCount(SPELL_TORCHES_CAUGHT) >= requiredCatches)
```

Comment:
What should happen if requiredCatches is 0 here ?

Reply:
Having the quest item without one of those quests' status incomplete should be impossible normally. The quest item is removed when the quest is completed. Unless a GM adds it to the player manually, in which case all bets are off I believe. Right now it'd let the quest complete regardless of the aura, which I think is fine for this not-normally-possible case :P

Reply:
I would

Reply:
could you please handle the case I mentioned ?